### ISSUE TITLE
ship only version with openssl

### ISSUE THREAD

**Issue by jumoog** [eclipse-paho/paho.mqtt.c#946]
Many beginners make the mistake of using the wrong library.
We should only release the version with ssl support.
Even if you build the library yourself, if you enable ssl support, you should only build the ssl variant. 
This should avoid frustration for future users.

**Comment by icraggs** [comment]
Yes, it's something I'd thought about too.  The easiest way in CMake is to make the "build with SSL" option the default and to not build the non-SSL versions unless another option such as "build non-SSL libraries" is chosen.
